index.js

```diff
@@ -81,7 +81,7 @@ function _parseInput (input, opts, cb) {
 
   var commonPrefix = null
   input.forEach(function (item, i) {
-    if (typeof item === 'string') return
+    if (typeof item === 'string' || (Buffer.isBuffer(item) && !item.name)) return
```

Comment:
Is there a reason for extending nameless Buffer/Streams with a name (e.g. `opts.name`) ?